### PR TITLE
Recursive indexing

### PR DESCRIPTION
**Pull request by Avokadoen**
Pull request for #11

A bit bad practice, but to avoid creating hundreds of PRs I also include some minor refactors

### CODE REVIEW

**Comment by Avokadoen** [comment]
Draft for issue #11

File: pkg/index/autoindex.go [hunk 1]
```diff
@@ -17,23 +17,26 @@
...
 )
 
 type autoindexer struct {
 	watcher     *fsnotify.Watcher
 	indexWorker *indexWorker
+	watchDepth  int
 }
 
-func NewAutoIndexer(db *Db) *autoindexer {
+func NewAutoIndexer(db *Db, watchDepth int) *autoindexer {
```
Comment (maeb) [inline pkg/index/autoindex.go:36]: warcdir slice as argument

File: pkg/index/autoindex.go [hunk 2]
```diff
@@ -73,30 +97,39 @@ func (a *autoindexer) fileWatcher() {
...
 }
 
-func (a *autoindexer) indexDir(dir string) {
-	f, err := os.Open(dir)
+// Recursively add directory to autoindexer watcher and index it.
+// This function will **panic** if path is a file or does not exist
+func (a *autoindexer) addAndIndexDir(path string, currentDepth int) {
+	err := a.watcher.Add(path)
 	if err != nil {
 		log.Fatal(err)
```
Comment (maeb) [inline pkg/index/autoindex.go:110]: Describe error

File: pkg/index/autoindex.go [hunk 3]
```diff
@@ -73,30 +97,39 @@ func (a *autoindexer) fileWatcher() {
...
 	}
+
+	f, err := os.Open(path)
+	if err != nil {
+		log.Fatal(err)
```
Comment (maeb) [inline pkg/index/autoindex.go:115]: Describe error

**Review by maeb — CHANGES_REQUESTED** [review]

File: pkg/index/autoindex.go [hunk 4]
```diff
@@ -33,94 +32,92 @@ type autoindexer struct {
...
-	<-done
 }
 
 // Recursively add directory to autoindexer watcher and index it.
 // This function will **panic** if path is a file or does not exist
 func (a *autoindexer) addAndIndexDir(path string, currentDepth int) {
 	err := a.watcher.Add(path)
 	if err != nil {
-		log.Fatal(err)
+		log.Fatalf("Failed to watch on path: %v, error: v%", path, err)
```
Comment (maeb) [inline pkg/index/autoindex.go:109]:
```
log.Fatalf("Failed to watch %v: %v", path, err)
```

File: pkg/index/autoindex.go [hunk 5]
```diff
@@ -33,94 +32,92 @@ type autoindexer struct {
...
 	}
 
 	f, err := os.Open(path)
 	if err != nil {
-		log.Fatal(err)
+		log.Fatalf("Failed to open file at: %v, error: %v", path, err)
```
Comment (maeb) [inline pkg/index/autoindex.go:114]:
```
log.Fatalf("Failed to open file %v: %v", path, err)
```

File: pkg/index/autoindex.go [hunk 6]
```diff
@@ -33,94 +32,92 @@ type autoindexer struct {
...
 	}
 
 	files, err := f.Readdir(-1)
-	f.Close()
 	if err != nil {
-		log.Fatalf("%v: %v", f.Name(), err)
+		log.Fatalf("%Failed to read directory. Name: %v, error: %v", f.Name(), err)
```
Comment (maeb) [inline pkg/index/autoindex.go:119]:
```
log.Fatalf("Failed to read directory %v: %v", f.Name(), err)
```

**Review by maeb — CHANGES_REQUESTED** [review]

**Review by maeb — APPROVED** [review]

**Pull request merged**